File: SampleApp/basefiles/src/main/java/com/rp/basefiles/BasePresenter.java
package com.rp.basefiles;

/**
 * Created by rahul on 4/1/18.
 */

public abstract class BasePresenter<V extends IBaseView> implements IBasePresenter<V> {

    private V view;

    @Override
    public void onAttach(V baseView) {
        this.view = baseView;
    }

    public V view() {
        return view;
    }

    @Override
    public void onDestroy() {
        view = null;
    }
}

File: basefiles/src/main/java/com/rp/basefiles/BasePresenter.kt
package com.rp.basefiles

abstract class BasePresenter<V : IBaseView> : IBasePresenter<V> {

    private var view: V? = null

    override fun onAttach(baseView: V) {
        this.view = baseView
    }

    fun view(): V? = view

    override fun onDestroy() {
        view = null
    }
}